Sources/Search/OASearchUICore.mm

```diff
@@ -201,13 +202,29 @@ -(NSComparisonResult) compare:(OASearchResult *)o1 o2:(OASearchResult *)o2 compa
             }
             else if (am1 && am2)
             {
+                // here 2 points are amenity
                 const auto& a1 = std::dynamic_pointer_cast<const OsmAnd::Amenity>(o1.amenity);
                 const auto& a2 = std::dynamic_pointer_cast<const OsmAnd::Amenity>(o2.amenity);
-                NSComparisonResult cmp = (NSComparisonResult)OsmAnd::ICU::ccompare(a1->type, a2->type);
+                
+                QString type1 = a1->type;
+                QString type2 = a2->type;
+                QString subType1 = a1->subType == nullptr ? "" : a1->subType;
```

Comment:
This check is not correct. QString cannot be nullptr because: c++ has default initialization (https://en.cppreference.com/w/cpp/language/default_initialization) and most importantly, Qstring is not a pointer so it cannot be nullptr. Simply remove this check